snap/naming/validate.go

```diff
@@ -174,3 +175,45 @@ func ValidateSnapID(id string) error {
 	}
 	return nil
 }
+
+var errInvalidSecurityTag = errors.New("invalid security tag")
+
+// ValidateSecurityTag validates known variants of snap security tag.
+//
+// Two forms are recognised, one for apps and one for hooks. Other forms
+// are possible but are not handled here.
+//
+// TODO: handle the weird udev variant.
+func ValidateSecurityTag(tag string) error {
+	parts := strings.SplitN(tag, ".", 4)
+	switch len(parts) {
+	case 3:
+		snapLiteral, snapName, appName := parts[0], parts[1], parts[2]
+		if snapLiteral != "snap" {
+			return errInvalidSecurityTag
+		}
+		if err := ValidateInstance(snapName); err != nil {
+			return errInvalidSecurityTag
+		}
```

Comment:
Again, not a blocker, but you could slightly refactor and put the snapLiteral and ValidateInstance() checks before the switch since they are identical for both case 3 and 4 (you'd then obviously have to check the length is at least 2 though).

Reply:
Good idea, applied.